netket/stats/_sum_inplace.py

```diff
@@ -78,7 +78,8 @@ def sum_inplace_jax(x):
             if _n_nodes == 1:
                 return x
             else:
-                res, _ = mpi4jax.Allreduce(x, op=_MPI.SUM, comm=_MPI_comm)
+                token = jax.lax.create_token(0)
+                res, _ = mpi4jax.Allreduce(x, op=_MPI.SUM, comm=_MPI_comm, token=token)
```

Comment:
@PhilipVinc This is the only way I could get it working, although i think that is because the token might be eliminated by dead code elimination?

Otherwise I either get  NotImplementedError: Transpose rule (for reverse-mode differentiation) for 'create_token' not implemented  if I dont pass a token or let it depend on x here, or TypeError: No constant handler for type: <class 'jax.interpreters.xla.Token'>  if I create the token outside of the function and just capture it here.

Reply:
This should be ok. 
In any case we don't need tokens for our usecase because there is only 1 mpi call and they are sequential.

However, I'll have to find a solution for mpi4jax in the future.